<a href="https://colab.research.google.com/github/WellBrian/FUTURE_DS_03/blob/main/Student_satisfaction_survey.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Import necessary libraries

In [1]:
from google.colab import drive
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

drive.mount('/content/drive')

Mounted at /content/drive


Load the file

In [11]:
file_path = "/content/Student_Satisfaction_Survey.csv"

df = pd.read_csv(file_path, encoding='latin1', index_col='SN')
df.head()

,Total Feedback Given,Total Configured,Questions,Weightage 1,Weightage 2,Weightage 3,Weightage 4,Weightage 5,Average/ Percentage,Course Name,Basic Course
SN,,,,,,,,,,,
1,1,12,How much of the syllabus was covered in the cl...,0,0,1,0,0,3.00 / 60.00,FY B.VOC FOOD TECHNOLOGY,B.VOC FOOD TECHNOLOGY
2,1,12,How well did the teachers prepare for the clas...,0,0,0,0,1,5.00 / 100.00,FY B.VOC FOOD TECHNOLOGY,B.VOC FOOD TECHNOLOGY
3,1,12,How well were the teachers able to communicate?,0,0,0,0,1,5.00 / 100.00,FY B.VOC FOOD TECHNOLOGY,B.VOC FOOD TECHNOLOGY
4,1,12,The teacher’s approach to teaching can best be...,0,0,1,0,0,3.00 / 60.00,FY B.VOC FOOD TECHNOLOGY,B.VOC FOOD TECHNOLOGY
5,1,12,Fairness of the internal evaluation process by...,0,0,0,1,0,4.00 / 80.00,FY B.VOC FOOD TECHNOLOGY,B.VOC FOOD TECHNOLOGY


In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 580 entries, 1 to 20
Data columns (total 11 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Total Feedback Given  580 non-null    int64 
 1   Total Configured      580 non-null    int64 
 2   Questions             580 non-null    object
 3   Weightage 1           580 non-null    int64 
 4   Weightage 2           580 non-null    int64 
 5   Weightage 3           580 non-null    int64 
 6   Weightage 4           580 non-null    int64 
 7   Weightage 5           580 non-null    int64 
 8   Average/ Percentage   580 non-null    object
 9   Course Name           580 non-null    object
 10  Basic Course          580 non-null    object
dtypes: int64(7), object(4)
memory usage: 54.4+ KB


The dataset contains information related to a survey conducted at an autonomous college to evaluate faculty performance. Here's a breakdown of the features in the dataset:

SN: Likely a serial number or identifier for each entry in the survey.  

Total Feedback Given: The number of feedback entries provided by students for a specific question.  

Total Configured: Possibly refers to the total number of students or responses expected for each question.  

Questions: The actual question asked in the survey, related to the performance and effectiveness of faculty members.  

Weightage 1 to Weightage 5: These fields could represent responses or ratings given to each question, possibly on a scale (like 1-5), or categories of answers.  

Average/ Percentage: This could be the average score or percentage calculated from the responses, providing a quantitative measure of the feedback for each question.  

Course Name: The specific course to which the feedback applies.  

Basic Course: A more general category or type of course under which the specific course falls.  

### **Descriptive Statistics**

In [12]:
df["Average"] = df["Average/ Percentage"].str.split("/").str[0].astype(float)
df['Average'].describe()

,Average
count,580.000000
mean,3.842793
std,0.629038
min,1.330000
25%,3.500000
50%,3.920000
75%,4.250000
max,5.000000


Here are the basic statistics for the average scores provided in the feedback:

Count: 580 responses

Mean (Average): 3.84

Standard Deviation: 0.63, indicating the variability of scores around the mean

Minimum Score: 1.33

25th Percentile: 3.50 (indicating that 25% of scores are below 3.50)

Median (50th Percentile): 3.92

75th Percentile: 4.25 (75% of scores are below this value)

Maximum Score: 5.00